File: ios/Packages/GemstonePrimitives/Sources/PerpetualFormatter.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation
import Gemstone
import Primitives

public struct PerpetualFormatter {
    private let perpetual: Gemstone.GemPerpetual

    public init(provider: Primitives.PerpetualProvider) {
        perpetual = Gemstone.GemPerpetual(provider: provider.toGem())
    }

    public func formatInputPrice(_ price: Double, decimals: Int32, locale: Locale = .current) -> String {
        perpetual.formatInputPrice(price: price, decimals: decimals, decimalSeparator: NumberInput.format(locale).decimalSeparator)
    }
}

File: android/gemcore/src/main/kotlin/com/gemwallet/android/ext/PerpetualFormatter.kt
package com.gemwallet.android.ext

import com.gemwallet.android.ext.toGem
import com.gemwallet.android.math.numberFormat
import com.wallet.core.primitives.PerpetualProvider
import uniffi.gemstone.GemPerpetual
import java.util.Locale

object PerpetualFormatter {

    fun formatPrice(provider: PerpetualProvider, price: Double, decimals: Int): String =
        GemPerpetual(provider.toGem()).use { it.formatPrice(price, decimals) }

    fun formatInputPrice(
        provider: PerpetualProvider,
        price: Double,
        decimals: Int,
        locale: Locale = Locale.getDefault(),
    ): String = GemPerpetual(provider.toGem()).use { it.formatInputPrice(price, decimals, numberFormat(locale).decimalSeparator) }
}
